doc/api/worker.md

```diff
@@ -0,0 +1,469 @@
+# Worker
+
+<!--introduced_in=REPLACEME-->
+
+> Stability: 1 - Experimental
+
+The `worker` module provides a way to create multiple environments running
+on independent threads, and to create message channels between them. It
+can be accessed using:
+
+```js
+const worker = require('worker');
+```
+
+Workers are useful for performing CPU-intensive JavaScript operations; do not
+use them for I/O, since Node.js’s built-in mechanisms for performing operations
+asynchronously already treat it more efficiently than Worker threads can.
+
+Workers, unlike child processes or when using the `cluster` module, can also
+share memory efficiently by transferring `ArrayBuffer` instances or sharing
+`SharedArrayBuffer` instances between them.
+
+## Example
+
+```js
+const { Worker, isMainThread, postMessage, workerData } = require('worker');
+
+if (isMainThread) {
+  module.exports = async function parseJSAsync(script) {
+    return new Promise((resolve, reject) => {
+      const worker = new Worker(__filename, {
+        workerData: script
+      });
+      worker.on('message', resolve);
+      worker.on('error', reject);
+      worker.on('exit', (code) => {
+        if (code !== 0)
+          reject(new Error(`Worker stopped with exit code ${code}`));
+      });
+    });
+  };
+} else {
+  const { parse } = require('some-js-parsing-library');
+  const script = workerData;
+  postMessage(parse(script));
+}
+```
+
+## Event: 'workerMessage'
+<!-- YAML
+added: REPLACEME
+-->
+
+If this thread was spawned by a `Worker`, this event emits messages sent using
+[`worker.postMessage()`][]. See the [`port.on('message')`][] event for
+more details.
+
+Listeners on this event will receive a clone of the `value` parameter as passed
+to `postMessage()` and no further arguments.
+
+This event is emitted on the `worker` module as returned by
+`require('worker')` itself.
```

Comment:
@addaleax .... have you looked into what it would take to support the more common `onmessage = function() {}` syntax with this? Using the EventEmitter is likely more Node.js like, but supporting `onmessage` as an option would be worthwhile from an isometric javascript point of view.

Reply:
it does support that

Reply:
Yes, it’s supported but not really documented at this point. The only real reason to have it is web compat, as you’re saying :)

Reply:
This listener-on-module design doesn’t seem to map very well to ESM…